.github/workflows/test.yml

```
@@ -48,3 +48,60 @@ jobs:
 
       - name: Run widget tests, unit tests
         run: flutter test --machine --coverage
+
+  generate_mason_bundle:
+    # The job generate_mason_bundle identify that the job test must complete successfully
+    # before this generate_mason_bundle job will run
+    needs: test
+    name: Generate Mason bundle
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+
+      - name: Set up Flutter environment
+        uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+          flutter-version: '3.10.5'
+
+      - name: Remove the old sample project
+        run: |
+          rm -rf sample
+
+      - name: Install Mason CLI
+        run: |
+          dart pub global activate mason_cli
+          mason get
+
+      - name: Generate Mason bundle
+        run: |
+          mason bundle bricks/permission_handler -t dart -o bricks/template/hooks/bundles
+
+      - name: Set the project name "sample"
+        uses: jossef/action-set-json-field@v2.1
+        with:
+          file: mason-config.json
+          field: project_name
+          value: sample
+
+      - name: Generate the new sample project
+        run: |
+          mason make template -c mason-config.json
+
+      - id: changes
+        name: Check for changes in the sample project
+        run: |
+          count=$(git status sample --porcelain | wc -l)
+          echo "count=$count" >> $GITHUB_OUTPUT
+
+      - name: Commit & push the sample project changes
+        if: steps.changes.outputs.count > 0
+        run: |
+          git config user.name team-nimblehq
+          git config user.email bot@nimblehq.co
+          git add bricks
+          git add sample
+          git commit -m "[Chore] Generate Mason bundle"
+          git push
```

and push changes for mason bricks

```suggestion
      - name: Commit & push the mason bundle changes
        if: steps.changes.outputs.count > 0
        run: |
          git config user.name team-nimblehq
          git config user.email bot@nimblehq.co
          git add bricks
          git commit -m "[Chore] Generate Mason bundle"
          git push
```

---

@luongvo I think if there any changes in the brick, then that change one way or another will be applied to the sample project. So that's why I think we should add change in sample project too. 🤔

---

@nmint8m The sample project generating will be improved later in a separate task https://github.com/nimblehq/flutter-templates/discussions/219.